crates/router/src/connector/stripe.rs

```diff
@@ -1026,23 +1027,70 @@ impl api::IncomingWebhook for Stripe {
             .parse_struct("StripeWebhookObjectId")
             .change_context(errors::ConnectorError::WebhookReferenceIdNotFound)?;
 
-        Ok(api_models::webhooks::ObjectReferenceId::PaymentId(
-            api_models::payments::PaymentIdType::ConnectorTransactionId(details.data.object.id),
-        ))
+        Ok(match details.data.object.object.as_str() {
```

Comment:
details.data.object.object doesn't tell what data you are accessing, give better naming here for each field